app/models/hubstats/deploy.rb

```diff
@@ -1,6 +1,17 @@
 module Hubstats
   class Deploy < ActiveRecord::Base
 
+    before_validation :check_time, on: :create
+    validates :git_revision, :deployed_at, :deployed_by, :repo_id, presence: true
+    validates_associated :repo
+    validates_presence_of :repo_id
```

Comment:
I think that the two of them check for separate things. validates_presence_of checks that repo_id isn't blank. validates_associated checks that the repo exists already.